### PR TITLE
upgrade traderjoe

### PR DESCRIPTION
**Pull request by vic-en**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
This pr upgrade TraderJoe to use sdk V2.
[ch41010]

**Tests performed by the developer**:
- Curl test 
- End-to-end testing

**Tips for QA testing**:
- Curl test 
- End-to-end testing

### CODE REVIEW

**Review by rxlxrxsx — APPROVED** [review]
Tested and confirmed working by QA.

- Successfully cloned and installed the branch
- Successfully setup gateway
- Tested the endpoints for trader joe using curl and confirmed working as expected
- Connected trader joe and confirmed working as expected
- Confirmed wallet and exchange balances are reflecting correct
- Created a amm_arb strategy and went through initial prompts successfully
- Started the strategy without issues, confirmed orders are created and cancelled based on config
- Trade information are correct based on history command output
- Compared trades in client and connector, no discrepancies found

File: src/connectors/traderjoe/traderjoe.ts [hunk 1]
```diff
@@ -111,7 +124,7 @@ export class Traderjoe implements Uniswapish {
    * Default time-to-live for swap transactions, in seconds.
    */
   public get ttl(): number {
-    return this._ttl;
+    return Math.floor(new Date().getTime() / 1000) + this._ttl;
```
Comment (petioptrv) [inline src/connectors/traderjoe/traderjoe.ts:127]: This is an interface method. We should keep this as true time to live getter, and create an additional helper `deadline` that will return `Math.floor(new Date().getTime() / 1000) + this.ttl`

**Review by petioptrv — CHANGES_REQUESTED** [review]

File: src/connectors/traderjoe/traderjoe.ts (hunk 1 above)
Comment (petioptrv) [reply src/connectors/traderjoe/traderjoe.ts:127]: This method is called by and passed to a client of the interface in such a way that mandates `ttl` returns a deadline instead for Uniswapish

**Review by petioptrv — APPROVED** [review]
Changes LGTM

**Comment by vic-en** [comment]
closing in favour of https://github.com/hummingbot/gateway/pull/131

**Pull request closed without merging**